cns/restserver/util.go

```diff
@@ -797,10 +829,16 @@ func (service *HTTPRestService) getPrimaryHostInterface(ctx context.Context) (*w
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to get interfaces from IMDS")
 		}
+
 		primary, err := wireserver.GetPrimaryInterfaceFromResult(res)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to get primary interface from IMDS response")
 		}
+
+		if !isPrivateIP(net.ParseIP(primary.PrimaryIP)) {
+			return nil, errors.Wrap(err, "only private IPv4 or IPv6 address can be used")
+		}
```

Comment:
customer can create vnet in public ip space as well.. we shouldn't have this check